_docs/src/documentation/rex_pipeline.md

```diff
@@ -105,40 +105,70 @@ It'll scan for all sharpmake files recursively in 3 different folders
 
 all these files are then passed on to sharpmake and executed.
 
-The generation script also has some config flags to enable or disable some behavior done in the generation step.
+Because the generation script is a wrapper around a sharpmake call, the script itself doesn't own the commandline arguments, these are owned by Sharpmake.
+We needed a way to let sharpmake communicate with the python script and back so that we don't have to maintain the same list of commandlines twice, once in sharpmake and once in the python script. We've achieved this by creating a `default_config.json` file that lives next to the sharpmake build sources
 
-| argument                  |  description 
-|--                         |--:            
-| -no_clang_tools           | disables clang tools in any post build command  
-| -unit_tests               |  also generates unit test projects 
-| -coverage                 |  adds code coverage flags to compilation steps 
-| -asan                     | adds code address sanitizer to compilation steps  
-| -ubsan                    | adds code undefined behavior sanitizer to compilation steps  
-| -fuzzy                    |  adds code fuzzy testing to compilation steps 
-| -sharpmake_args           | pass in additional arguments to sharpmake  
+This config file is parsed by the python script and it'll create commandline argument for them. the script will create a `config.json` file in the intermediate build folder which is passed to sharpmake. It'll read the config file and use it to initialize the generation pipeline.
+
+There's only 1 extra commandline argument supported by the generate script which is `-sharpmake_args`. This is to allow a user to pass in arguments that are supported by the native sharpmake executable
+
+We support Visual Studio and Visual Studio Code as IDEs. These just act as text editors as they're in turn call into the build pipeline and build the ninja files which are always generated.
 
 Some Examples (Windows):
 ```sh
 # Default generation. generate files for engine and editor
-py _generate.py
+py _rex.py generate
 
 # Default generation + unit tests
-py _generate.py -unit_tests 
+py _rex.py generate -enable-unit-tests 
 
 # Default generation + unit tests + code coverage
-py _generate.py -unit_tests -coverage 
+py _rex.py generate -enable-unit-tests -enable-code-coverage 
+
+# Default generation + unit tests + code coverage + Visual Studio sln and projects
+py _rex.py generate -enable-unit-tests -enable-code-coverage -IDE VisualStudio
 
-# Default generation + disable clang tools. this skips the argument in the script and passes it directly to sharpmake 
-py _generate.py -sharpmake_args="/noClangTools" 
+# Default generation, but single threaded
+py _rex.py generate -sharpmake_args /multithreaded(false)
 ```
 
+### Code Generation
```

Comment:
Can you show an example of this as well, is this a separate command, is it done automatically when you call generate?